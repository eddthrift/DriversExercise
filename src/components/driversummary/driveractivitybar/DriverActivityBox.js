import { Box } from '@mui/material';
import {IsDriverActive} from '../../../utils/DriverUtils'
import React, {useState, useEffect} from 'react';

import './DriverActivityBar.css';

function DriverActivityBox(props){
    const [isActive, setIsActive] = useState(false);
    
    useEffect(() => 
    {
        setIsActive(IsDriverActive(props.driver, props.date));
    },[props.driver, props.date]);

    let className = isActive ? "activity-box-active" : "activity-box-idle";

    return(
        <Box className={className} sx={{border: 1}}/>
    );
};

export default DriverActivityBox;