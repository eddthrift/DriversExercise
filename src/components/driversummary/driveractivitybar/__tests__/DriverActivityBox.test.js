import React from 'react';
import renderer from 'react-test-renderer';

import DriverActivityBox from '../DriverActivityBar';

it('renders correctly', () => {
    const tree = renderer.create(<DriverActivityBox />).toJSON();
    expect(tree).toMatchSnapshot();
});