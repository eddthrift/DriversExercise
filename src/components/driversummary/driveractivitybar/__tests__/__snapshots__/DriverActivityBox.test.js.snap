// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiStack-root css-1xcvc1q-MuiStack-root"
>
  <div
    className="activity-box-idle MuiBox-root css-b7rhyr"
  />
  <div
    className="activity-box-idle MuiBox-root css-b7rhyr"
  />
  <div
    className="activity-box-idle MuiBox-root css-b7rhyr"
  />
  <div
    className="activity-box-idle MuiBox-root css-b7rhyr"
  />
  <div
    className="activity-box-idle MuiBox-root css-b7rhyr"
  />
  <div
    className="activity-box-idle MuiBox-root css-b7rhyr"
  />
  <div
    className="activity-box-idle MuiBox-root css-b7rhyr"
  />
</div>
`;
